vespa/deployment.py

```diff
@@ -550,127 +646,185 @@ def deploy(
         print("Finished deployment.", file=self.output)
         return app
 
-    def _get_latest_run_id(self, instance) -> int:
-        # The following endpoint returns a dictionary containing information about various builds for a given application.
-        # It sometimes takes a couple of seconds for the actual latest build to show up, but once it does, we can get the latest run id.
-        endpoint = f"/application/v4/tenant/{self.tenant}/application/{self.application}/deployment"
-        res = self._request("GET", endpoint)
-
-        # The different deployment stages might be out of sync, so we need all the ids to determine the latest one
-        run_ids = []
-        for item in res["steps"]:
-            if (
-                "runs" in item.keys() and len(item["runs"]) > 0
-            ):  # "id" is only present in steps with "runs" key
-                run_ids.append(item["runs"][0]["id"])  # Index zero to get the latest id
-        if run_ids == []:
-            return -1  # No runs found
-
-        return max(run_ids)
-
-    def _get_run_id_with_retry(self, instance, last_run_id):
-        run_id = self._get_latest_run_id(instance)
-        if run_id <= last_run_id:
-            raise Exception("Run id not updated yet.")
-        return run_id
-
-    @run_with_max_time
     def deploy_to_prod(
         self,
         instance: Optional[str] = "default",
         disk_folder: Optional[str] = None,
-        max_wait: int = 600,
+        submit_options: Dict[str, str] = {},
     ) -> None:
         """
         Deploy the given application package as the given instance in the Vespa Cloud prod environment.
         NB! This feature is experimental and may fail in unexpected ways. Expect better support in future releases.
 
         :param instance: Name of this instance of the application, in the Vespa Cloud.
-        :param disk_folder: Disk folder to save the required Vespa config files. Default to application name
-            folder within user's current working directory.
-        :param max_wait: Seconds to wait for the deployment.
+        :param disk_folder: Disk folder to either save the required Vespa config files (if initialized with application_package) or read them from (if initialized with disk_folder).
+        :param submit_options: Optional dictionary of submit options, containing one or more of the following keys:
+            - "repository"
+            - "branch"
+            - "commit"
```

Comment:
These are somewhat obsolete with `sourceUrl`, which is otherwise generated from these three on the server side. I think it's sufficient to just expose that one parameter.

Reply:
Aha, will only expose `sourceUrl`then!